src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
+
+	public void resetBasket() {
+		buyBasket();
+		codeDePromotion = false;
+	}
```

Comment:
After you solve the other comments, perhaps we need a short call to discuss the concept of code of promotion.

Since boolean might not be enough in the long term if business decide to apply different coupons to the basket.